File: backend/app/schemas/brand_library.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import LibraryItemStatus, ArtifactType


class LibraryArtifactResponse(BaseModel):
    id: uuid.UUID
    name: str
    type: ArtifactType
    thumbnail_url: str | None = None
    product: str | None = None

    model_config = {"from_attributes": True}


class LibraryPublisherResponse(BaseModel):
    id: uuid.UUID
    name: str

    model_config = {"from_attributes": True}


class BrandLibraryItemResponse(BaseModel):
    id: uuid.UUID
    artifact: LibraryArtifactResponse
    published_by: LibraryPublisherResponse
    status: LibraryItemStatus
    remix_count: int
    published_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class BrandLibraryListResponse(BaseModel):
    items: list[BrandLibraryItemResponse]
    total: int
    page: int
    per_page: int

File: frontend/src/types/brand-library.ts
export type LibraryItemStatus = "pending_review" | "approved" | "published" | "rejected";

export interface LibraryArtifact {
  id: string;
  name: string;
  type: string;
  thumbnail_url: string | null;
  product: string | null;
}

export interface LibraryPublisher {
  id: string;
  name: string;
}

export interface BrandLibraryItem {
  id: string;
  artifact: LibraryArtifact;
  published_by: LibraryPublisher;
  status: LibraryItemStatus;
  remix_count: number;
  published_at: string | null;
  created_at: string;
}

export interface BrandLibraryListResponse {
  items: BrandLibraryItem[];
  total: number;
  page: number;
  per_page: number;
}
